cli/dcoscli/auth/main.py

```diff
@@ -68,34 +75,139 @@ def _info(info):
     return 0
 
 
-def _login():
+def _list_providers(json_):
     """
-    :returns: process status
+    :returns: providers available for configured cluster
+    :rtype: dict
+    """
+
+    providers = auth.get_providers()
+    if providers or json_:
+        emitting.publish_table(
+            emitter, providers, tables.auth_provider_table, json_)
+    else:
+        raise DCOSException("No providers configured for your cluster")
+
+
+def _get_password(password_str, password_env, password_file):
+    """
+    Get password for authentication
+
+    :param password_str: password
+    :type password_str: str
+    :param password_env: name of environment variable with password
+    :type password_env: str
+    :param password_file: path to file with password
+    :type password_file: bool
+    :returns: password or None if no password specified
+    :rtype: str | None
+    """
+
+    password = None
+    if password_str:
```

Comment:
the user would have to specify `--password_str=""` to input an empty string, but I guess thats valid.